pkg/render/istio/istio.go

```diff
@@ -0,0 +1,384 @@
+	}
+	return nil
+}
+
+func (c *IstioComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	var err error
+
+	reg := c.cfg.Installation.Registry
+	path := c.cfg.Installation.ImagePath
+	prefix := c.cfg.Installation.ImagePrefix
+
+	c.IstioPilotImage, err = components.GetReference(components.ComponentCalicoIstioPilot, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	c.IstioInstallCNIImage, err = components.GetReference(components.ComponentCalicoIstioInstallCNI, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	c.IstioZTunnelImage, err = components.GetReference(components.ComponentCalicoIstioZTunnel, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	c.IstioProxyv2Image, err = components.GetReference(components.ComponentCalicoIstioProxyv2, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	if err = c.patchImages(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Objects implements the Component interface.
+func (c *IstioComponent) Objects() ([]client.Object, []client.Object) {
+	res := c.resources
+
+	objs := []client.Object{}
+	objs = append(objs,
+		c.istiodAllowTigeraPolicy(),
+		c.istioCNIAllowTigeraPolicy(),
+		c.ztunnelAllowTigeraPolicy(),
+	)
+
+	if overrides := c.cfg.Istio.Spec.IstiodDeployment; overrides != nil {
+		rcomp.ApplyDeploymentOverrides(res.IstiodDeployment, overrides)
+	}
+
+	if overrides := c.cfg.Istio.Spec.IstioCNIDaemonset; overrides != nil {
+		rcomp.ApplyDaemonSetOverrides(res.CNIDaemonSet, overrides)
+	}
+
+	if overrides := c.cfg.Istio.Spec.ZTunnelDaemonset; overrides != nil {
+		rcomp.ApplyDaemonSetOverrides(res.ZTunnelDaemonSet, overrides)
+	}
+
+	// Set required configs
+	for i := range res.ZTunnelDaemonSet.Spec.Template.Spec.Containers {
+		cont := &res.ZTunnelDaemonSet.Spec.Template.Spec.Containers[i]
+		if cont.Name == "istio-proxy" {
+			cont.Env = append(cont.Env, corev1.EnvVar{
+				Name:  "TRANSPARENT_NETWORK_POLICIES",
+				Value: "true",
+			})
+			break
+		}
+	}
+	for i := range res.CNIDaemonSet.Spec.Template.Spec.Containers {
+		cont := &res.CNIDaemonSet.Spec.Template.Spec.Containers[i]
+		if cont.Name == "install-cni" {
+			cont.Env = append(cont.Env, corev1.EnvVar{
+				Name:  "MAGIC_DSCP_MARK",
+				Value: strconv.FormatInt(int64(c.cfg.Istio.Spec.DSCPMark.ToUint8()), 10),
+			})
+		}
+	}
+
+	// Set additional pull secrets
+	res.IstiodDeployment.Spec.Template.Spec.ImagePullSecrets = append(
+		res.IstiodDeployment.Spec.Template.Spec.ImagePullSecrets,
+		secret.GetReferenceList(c.cfg.PullSecrets)...,
+	)
+	res.CNIDaemonSet.Spec.Template.Spec.ImagePullSecrets = append(
+		res.CNIDaemonSet.Spec.Template.Spec.ImagePullSecrets,
+		secret.GetReferenceList(c.cfg.PullSecrets)...,
+	)
+	res.ZTunnelDaemonSet.Spec.Template.Spec.ImagePullSecrets = append(
+		res.ZTunnelDaemonSet.Spec.Template.Spec.ImagePullSecrets,
+		secret.GetReferenceList(c.cfg.PullSecrets)...,
+	)
+
+	// Append Istio resources in order: Base, Istiod, CNI, ZTunnel
```

Comment:
Is there a reason they need to be in this order? If so, good to call it out explicitly. Otherwise, we can just delete this comment as it suggests the order is important.